CONTRIBUTING.md

```diff
@@ -1,80 +1,96 @@
 
-At this stage one should expect comments and review from other contributors. You can add more commits to your pull request by committing them locally and pushing to your fork until you have satisfied all feedback. If your pull request is accepted for merging, you may be asked by a maintainer to squash and or rebase your commits before it will be merged. The length of time required for peer review is unpredictable and will vary from patch to patch.
+## <a name="codeofconduct"></a> Code of Conduct
 
+We want a productive, happy and agile community that can welcome new ideas in a complex field, improve every process every year, and foster collaboration between groups with very different needs, interests and skills. We gain strength from diversity, and actively seek participation from those who enhance it. This code of conduct exists to ensure that diverse groups collaborate to mutual advantage and enjoyment. We will challenge prejudice that could jeopardise the participation of any person in the project.
 
-Pull Request Philosophy
------------------------
+The Code of Conduct governs how we behave in public or in private whenever the project will be judged by our actions. We expect it to be honoured by everyone who represents the project officially or informally, claims affiliation with the project, or participates directly.
 
-Patchsets should always be focused. For example, a pull request could add a feature, fix a bug, or refactor code; but not a mixture. Please also avoid super pull requests which attempt to do too much, are overly large, or overly complex as this makes review difficult.
+**We strive to:**
 
+- Be considerate
+	- Our work will be used by other people, and we, in turn, will depend on the work of others. Any decision we take will affect users and colleagues, and we should consider them when making decisions.
 
-###Features
+- Be respectful
+	- Disagreement is no excuse for poor manners. We work together to resolve conflict, assume good intentions and do our best to act in an empathic fashion. We don't allow frustration to turn into a personal attack. A community where people feel uncomfortable or threatened is not a productive one.
 
-When adding a new feature, thought must be given to the long term technical debt and maintenance that feature may require after inclusion. Before proposing a new feature that will require maintenance, please consider if you are willing to maintain it (including bug fixing). If features get orphaned with no maintainer in the future, they may be removed by the Repository Maintainer.
+- Take responsibility for our words and our actions
+	- We can all make mistakes; when we do, we take responsibility for them. If someone has been harmed or offended, we listen carefully and respectfully, and work to right the wrong.
```

Comment:
Suggestion: "We can all make mistakes; when we do, we take responsibility for them. If we make a claim that turns out to untrue, we correct it. If someone has been harmed or offended, we listen carefully and respectfully, and work to right the wrong." Overall, I think these are great, but I'm hoping to get a little dose of "We speak truthfully".